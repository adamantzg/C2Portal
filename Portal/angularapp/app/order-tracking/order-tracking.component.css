

.tracking-map-holder {
  padding-top: 45px;
  /* margin-left: 15%; */
}

.order-number {
  /* padding-bottom: 15px; */
  padding-left: 40px;
}

.tracking {
  padding-top: 15px;
  padding-bottom: 15px;
  text-align: center;
  /* background: rgba(0, 128, 0, .3); */
  /* color: rgba(0, 128, 0, 1); */
}

.number-info {
  /* bottom: 40px;*/
  margin-top: 90px;
  margin-left: 40px;
  /* position: absolute; */
  /* position: absolute; */
  /* margin-top: 20%; */
  /* align-self:flex-end; */
}

.tracking-name,
.tracking-title {
  text-align: center;
}

.responsive-circle {
  font-size: 80px;
  font-weight: 700;
  margin: 0 auto 20px;
  width: 100%;
}

.panel-shipping{
    padding-bottom: 40px;
    padding-top:0;
}
.shippin-holder-text{
    width:264px;
    float:left;
}

.circle {
  /* border: 1px solid black; */
  float: left;
  width: 150px;
  height: 150px;
  display: block;
  border-radius: 50%;
  font-size: 70px;
  margin-top: 20px;
  /* background: #FEAEB0; */
}

.circle-holder span {
  position: absolute;
  bottom: 31px;
  width: 150px;
  margin-left: -150px;
  text-align: center;
}



.arrow-holder.recived,
.arrow-holder.processed {
 
  background: none;
}

.row-margin-top{
  margin-top:40px;
}



/* .glyphicon, */
.circle>.fa {
  margin-left: 35px;
  font-size: 70px;
  /* color: white; */
  margin-top: 35px;
}

.arrow-holder {
  float: left;
  /* color: #FEAEB0; */
  margin-top: 50px;
  margin-left: 25px;
  margin-right: 20px;
}

.shipping-holder {
  /* position: relative; */
  width: 800px;
  margin-left: auto;
  margin-right: auto;
}

.address {
  float: left;
  padding-right: 10px;
  min-width: 70px;
}

.flex,
.flex>div[class*='col-'] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex: 1;
}

.panel-height {
  height: 250px
}

.panel-heading-border{
  border-bottom:1px solid rgba(44, 62, 80,.6);
  padding-bottom:5px;
}
.list-group-item {
  background: none!important;
  border:none!important;
  padding:0!important;
}
.no-padding{
  padding:0!important;
}
.list-group-item {
  padding-left: 20px!important;
  /* height: 40px; */
}
.col-track{
  padding-right:0;
  height:172px;
}
.col-track>span{
  position:absolute;
  top:75px;
}
@media only screen and (min-width : 768px) {
            .responsive-circle{
                font-size: 50px;
            }
        }

@media only screen and (min-width: 1200px) {
  .responsive-circle {
    font-size: 65px;
        /* font-weight: 700;
        margin: 0 auto 20px;  */
    /* width: 70%; */
  }
}
@media screen and (max-width: 768px) {
  .row-margin-top{
    margin-top:0px;
  }
  .panel-heading.visible-xs{
    padding-right:0;
    padding-left:0;
  }

  .col-xs-12.visible-xs> .row > .col-xs-6 {
    /* border:1px solid green; */
    min-width: 180px;
  }
  .fa-xs-holder{
    width:100%;
    padding:5px 0 0 65px;
    float: left;
  }
  .fa-xs-holder> i{
    font-size:40px;
    position:absolute;
  }
  /* .col-xs-12.visible-xs>.row {

  } */
  .col-xs-12.visible-xs > .row{
    min-width: 315px;
  }
}

.alert-danger {
  background:#95a5a6;
}
.text-danger{
  color:#95a5a6;
}