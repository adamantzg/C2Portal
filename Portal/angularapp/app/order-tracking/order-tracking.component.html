<div class="container-fluid">
  <br>
  <div class="row">
    <div class="col-sm-4 col-md-1">
      <button class="btn btn-default" (click)="navigateToCaller()">Back</button>
    </div>
    <div class="col-sm-8 col-md-11">
      <app-impersonatebox (CustomerChange)="showForCustomer($event)" *ngIf="user.isTopAdmin || user.isBranchAdmin"></app-impersonatebox>
    </div>
  </div>
  
  
  <div class="row" style="margin-top:-20px; margin-left:-26px; margin-right:-30px; text-align: left">
    <div class="col-sm-12">
      <div class="panel-default">
        <div class="panel-heading" style="padding-left: 27px!important;">
          <div class="form-inline">
            <div class="form-group">
              <label class="m-r-xs">C2 Order no*:</label>
              <input type="text" class="form-control" [(ngModel)]="order_no" (keypress)="onOrderInput($event)"> 
              <button class="btn btn-default m-l-xs" (click)="searchOrder()" [disabled]="!order_no">Search</button>
            </div>
          </div>
        </div>
      </div>
      
    </div>
  </div>
  <div class="row">
    <div class="col-sm-12">
      <div>*Order number can be obtained from your sales order acknowledgement</div>
    </div>
  </div>
  <div class="alert alert-danger" *ngIf="errorMessage">{{errorMessage}}</div>
  <div class="row row-margin-top" *ngIf="order">
    <div class="col-sm-12">
      <!-- <div class="panel" [ngClass]= "{'panel-info' : !order.date_entered , 'panel-warning' : (!!order.date_entered && !order.invoice_date) , 'panel-success' : !!order.invoice_date}" > -->
      <div class="panel">
        <div class="panel-heading hidden-xs">

          <div class="shipping-holder">
            &nbsp;
            <!-- <span class="shippin-holder-text">Shipped Via: </span> -->
            <span class="shippin-holder-text">Status:
              <strong>{{getStatus()}}</strong>
            </span>
            <span class="shippin-holder-text" *ngIf="+order.status != orderStatuses.Delivered && expectedDeliveryDate">Expected Delivery:
              <strong>{{expectedDeliveryDate|date: datePattern}}</strong>
            </span>
          </div>
        </div>
        <div class="panel-heading visible-xs">
          <div class="col-xs-12">
            <!-- <span class="h5">Shipped Via: </span> -->
          </div>
          <div class="col-xs-12">
            <span class="h5">Status: {{getStatus()}}</span>
          </div>
          <div class="col-xs-12" *ngIf="order.status != orderStatuses.Delivered && expectedDeliveryDate">
            <span class="h5">Expected Delivery: <strong>{{expectedDeliveryDate|date: datePattern}}</strong></span>
          </div>
        </div>

        <div class="panel-body panel-shipping" style="padding-right:0" *ngIf="order.status != orderStatuses.Cancelled">
        <!-- <div class="col-sm-3">-</div> -->
          <div class="col-xs-12 visible-xs" style="padding:0;">
            <div class="row" >
              <div class="col-xs-6">
                <div class="circle" [ngClass]="{'alert-danger' : +order.status < 1, 'alert-success': +order.status >= 1 }">
                  <i class="fa fa-shopping-cart" aria-hidden="true"></i>
                </div>
                <div class="fa-xs-holder"  
                [ngClass]="{'text-danger' : +order.status === 0,
                'text-warning' : +order.status >=1 && +order.status <= orderStatuses.Received, 'text-success': +order.status > orderStatuses.Received}">
                  &nbsp;
                  <i class="fa fa-long-arrow-down"></i>
                </div>
              </div>
              <div class="col-xs-5 col-track">
                <span>Order Received</span>
              </div>
            </div>

            <div class="row"  >
              <div class="col-xs-6">
                <div class="circle" [ngClass]="{'alert-danger':+order.status <= orderStatuses.Received, 'alert-success' : +order.status > orderStatuses.Received }">
                  <i class="fa fa-truck fa-5x fa-flip-horizontal" aria-hidden="true"></i>
                </div>
                <div class="fa-xs-holder" [ngClass]="{'text-danger' : +order.status <= 5,
                'text-warning' : +order.status > orderStatuses.Received && +order.status < orderStatuses.Delivered,
                 'text-success': +order.status === orderStatuses.Delivered}">
                  &nbsp;
                  <i class="fa fa-long-arrow-down"></i>
                </div>
              </div>
              <div class="col-xs-5 col-track">
                <span>Order Processed</span>
              </div>
            </div>

            <div class="row"  >
              <div class="col-xs-6">
                <div class="circle" [ngClass]="{'alert-danger': +order.status !== orderStatuses.Delivered,
                 'alert-success': +order.status === orderStatuses.Delivered}">
                  <i class="fa fa-check" aria-hidden="true"></i>
                </div>
              </div>
              <div class="col-xs-5 col-track">
                <span>Delivered</span>
              </div>
            </div>

          </div>

        </div>
       
      </div>
      <div class="panel-body panel-shipping hidden-xs" *ngIf="order.status !== orderStatuses.Cancelled" >
        <div class="shipping-holder">
          <div class="circle-holder">
            <div class="circle" [ngClass]="{'alert-danger' : +order.status < 1, 'alert-success': +order.status > 0  }">
              <i class="fa fa-shopping-cart" aria-hidden="true"></i>
            </div>
            <span>Order Received</span>
          </div>
          <div class="arrow-holder" 
          [ngClass]="{'text-danger' : +order.status === 0,
              'text-warning' : +order.status >=1 && +order.status <= orderStatuses.Received, 'text-success': +order.status > orderStatuses.Received}">
            <i class="fa fa-long-arrow-right fa-5x" aria-hidden="true"></i>
          </div>
          <div class="circle-holder">
            <div class="circle" [ngClass]="{'alert-danger':order.status <= orderStatuses.Received, 'alert-success' : order.status > orderStatuses.Received }">
              <i class="fa fa-truck fa-5x fa-flip-horizontal" aria-hidden="true"></i>
            </div>
            <span>Order processed</span>
          </div>
          <div class="arrow-holder" [ngClass]="{'text-danger' : +order.status <= 5,
          'text-warning' : +order.status > orderStatuses.Received && +order.status < orderStatuses.Delivered,
           'text-success': +order.status === orderStatuses.Delivered}">
            <i class="fa fa-long-arrow-right fa-5x" aria-hidden="true"></i>
          </div>
          <div class="circle-holder">
            <div class="circle" [ngClass]="{'alert-danger': +order.status !== orderStatuses.Delivered, 'alert-success': +order.status === orderStatuses.Delivered}">
              <i class="fa fa-check" aria-hidden="true"></i>
            </div>
            <span>Order delivered</span>
          </div>

        </div>
      </div>
      <div class="panel-body panel-shipping" *ngIf="order.status === orderStatuses.Cancelled" style="text-align:center">
        
        <div class="alert alert-danger">
           <h3>Order Cancelled</h3>
        </div>
      </div>
      <div class="panel-header">
        &nbsp;
      </div>
    </div>
  </div>
</div>

<div class="row row-margin-top" *ngIf="order">
  <div class="col-sm-1">&nbsp;</div>
  <div class="col-sm-5">
    <div class="panel panel-body">
      <div class="panel-heading panel-heading-border">
        <span class="panel-title">Order History</span>
      </div>
      <div class="panel-heading">
        <div class="list-group list-group-bg">

          <div *ngIf="order.invoice_date" class="list-group-item">
            Order Delivered: {{order.invoice_date | date : datePattern}}
          </div>
          <div *ngIf="getProcessedDate(order)" class="list-group-item">
            Order Processed: {{getProcessedDate(order) | date : datePattern}}
          </div>
          <div *ngIf="order.date_entered" class="list-group-item">
            Order Received: {{order.date_entered | date: datePattern }}
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class="col-sm-5">
    <div class="panel panel-body">
      <div class="panel-heading panel-heading-border">
        <span class="panel-title">Delivery Information</span>
      </div>
      <div class="panel-heading">
        <div class="list-group list-group-bg">
          <a class="list-group-item">
            <div class="col-sm-6 col-lg-4 no-padding">
              <i class="fa fa-user" aria-hidden="true"></i>&nbsp;&nbsp;Name: </div>

            <div class="col-sm-6 no-padding">
              <span>{{order.address1}}</span>
            </div>
          </a>

          <a class="list-group-item" style="display:block">
            <div class="col-sm-6 col-lg-4 no-padding">
              <i class="fa fa-home" aria-hidden="true"></i> Address:</div>
            <div class="col-sm-6 no-padding">{{order.address2}} </div>
          </a>
          <a class="list-group-item">
            <div class="col-sm-6 col-sm-push-6 col-lg-push-4 no-padding"> {{order.address3}} {{order.address4}} </div>
          </a>

          <!-- <a class="list-group-item">
              <div class="col-sm-6 col-sm-push-4">{{order.address5}}</div>
            </a> -->
          <!-- <a class="list-group-item">
              <div class="col-sm-6 col-sm-push-4">{{order.address6}}</div>
            </a> -->
          <br>
        </div>
      </div>
    </div>
  </div>
  
</div>
<app-block-ui></app-block-ui>
