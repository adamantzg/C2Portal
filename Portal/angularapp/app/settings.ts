export class Settings {
    static datePattern = 'dd MMMM yyyy';
    static maintitle= '';
    static apiRoot= '/api/';
}
