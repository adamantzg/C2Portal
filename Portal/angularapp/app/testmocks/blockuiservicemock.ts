import { EventEmitter } from '@angular/core';

export class BlockUIServiceMock {
    blockTicks = 0;
    unBlockTicks = 0;
    blockUIEvent: any = new EventEmitter<any>();

    startBlock() {
      this.blockTicks++;
    }

    stopBlock() {
      this.unBlockTicks++;
    }
}
