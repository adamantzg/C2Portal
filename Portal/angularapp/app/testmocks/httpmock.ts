import { of } from 'rxjs';

export class HttpMock {
    url: string;
    data: any;
    params: any;
    options: any;
    postResult: any;
    getResult: any;


    post<T>(url, data, params) {
      this.data = data;
      this.url = url;
      this.params = params;
      return of(this.postResult);
    }

    get<T>(url, options) {
        this.url = url;
        this.options = options;
        return of(this.getResult);
    }

  }
