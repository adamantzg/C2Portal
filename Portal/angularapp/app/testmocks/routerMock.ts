import { Component } from '@angular/core';

export class RouterMock {
    navigateUrls: string[];
    params: any;

    navigate(urls: string[], params?: any) {
        this.navigateUrls = urls;
        this.params = params;
    }
}

export class ActivatedRouteMock {

}


export class RouterLinkMock {

}

@Component({
    template: ''
  })
  export class DummyComponent {
  }
