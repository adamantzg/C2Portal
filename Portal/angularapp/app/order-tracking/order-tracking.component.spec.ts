import { OrderTrackingComponent } from './order-tracking.component';
import { UserService } from '../services/user.service';
import { OrderService } from '../services/order.service';
import { HttpClient } from '@angular/common/http';
import { HttpMock } from '../testmocks/httpmock';
import { HolidayService } from '../services/holiday.service';
import { RouterTestingModule } from '@angular/router/testing';
import { routes } from '../routestest';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { User, Order, Customer, OrderStatus, StockData } from '../domainclasses';
import { CommonAppModule } from 'common_components/common.module';
import { BlockUiComponent } from '../blockui/block-ui.component';
import { BlockUIService } from 'common_components/services/block-ui.service';
import { BlockUIServiceMock } from '../testmocks/blockuiservicemock';
import { ImpersonateboxComponent } from '../common/impersonatebox/impersonatebox.component';
import { FormsModule } from '@angular/forms';
import { TypeaheadModule } from 'ngx-bootstrap';
import { HttpService } from '../services/http.service';
import { CommonService } from '../../c2-flex/services/common.service';
import { ActivatedRoute, Router } from '@angular/router';
import { of } from 'rxjs';
import * as moment from 'moment';
import * as $ from 'jquery';
import { RouterMock } from '../testmocks/routerMock';
import { By } from '@angular/platform-browser';
import { exec } from 'child_process';
import { containsElement } from '@angular/animations/browser/src/render/shared';
import { Settings } from '../settings';

describe('order tracking', () => {

    let fixture: ComponentFixture<OrderTrackingComponent>;
    let component: OrderTrackingComponent;
    let userService, orderService;

    setupTestBed({
        declarations: [OrderTrackingComponent, BlockUiComponent, ImpersonateboxComponent],
        providers: [UserService, OrderService, HttpService,  {provide: HttpClient, useClass: HttpMock}, HolidayService,
            {provide: BlockUIService, useClass: BlockUIServiceMock}, CommonService],
        imports: [ RouterTestingModule.withRoutes(routes), CommonAppModule, FormsModule, TypeaheadModule.forRoot()]
    });

    beforeEach(() => {
        const u = new User();
        u.id = 1;
        u.isTopAdmin = false;
        u.isBranchAdmin = false;
        userService = TestBed.get(UserService);
        jest.spyOn(userService, 'CurrentUser', 'get').mockImplementation(() => u);
        orderService = TestBed.get(OrderService);

        fixture = TestBed.createComponent(OrderTrackingComponent);
        component = fixture.componentInstance;
    });

    test('should create', () => {
        expect(component).toBeTruthy();
        expect(fixture).toMatchSnapshot();
    });

    test('checkParams', () => {
        const route = TestBed.get(ActivatedRoute);
        route.queryParams = of({return_to: 'return', order_no: 'order'});
        const fn = jest.spyOn(component, 'searchOrder').mockImplementation(() => {});
        component.checkParams();
        expect(component.return_to).toBe('return');
        expect(component.param).toBe('order');
        expect(component.order_no).toBe('order');
        expect(fn.mock.calls.length).toBe(1);

        route.queryParams = of({order_no: 'order'});
        component.checkParams();
        expect(component.return_to).toBe('invoices');
        expect(component.param).toBe('order');
        expect(component.order_no).toBe('order');
        expect(fn.mock.calls.length).toBe(2);
        fn.mockReset();
    });

    test('searchOrder', () => {
        const returned = { result: new Order()};
        returned.result.customer = 'cust';
        const fn1 = jest.spyOn(orderService, 'getOrder').mockImplementation(() => of(returned) );
        const fn2 = jest.spyOn(component, 'getCustomer').mockImplementation(() => {});
        component.customer_code = 'cust';
        component.order_no = 'order  ';
        component.searchOrder();
        expect(fn1.mock.calls[0][0]).toBe('order');
        expect(fn1.mock.calls[0][1]).toBe(component.customer_code);
        expect(component.order).toBe(returned.result);
        expect(fn2.mock.calls[0][0]).toBe(returned.result.customer);

        returned.result = null;
        component.searchOrder();
        expect(component.errorMessage).toBe('Order not found or you don\'t have access.');
        expect(fn2.mock.calls.length).toBe(1);
        fn1.mockReset();
        fn2.mockReset();

    });

    test('getCustomer', () => {
        const returned = new Customer();
        returned.code = 'code';
        component.order = new Order();
        component.order.status = OrderStatus.Delivered;
        const fn1 = jest.spyOn(orderService, 'getCustomer').mockImplementation(() => of(returned));
        const fn2 = jest.spyOn(component, 'getExpectedDelivery').mockImplementation(() => {});
        component.getCustomer('cust');
        expect(component.customer).toBe(returned);
        expect(fn1.mock.calls[0][0]).toBe('cust');
        expect(fn2.mock.calls.length).toBe(0);

        component.order.status = OrderStatus.InProgress;
        component.getCustomer('cust');
        expect(fn2.mock.calls.length).toBe(1);
        fn1.mockReset();
        fn2.mockReset();

    });

    test('getExpectedDelivery', () => {
        const holidayService = TestBed.get(HolidayService);
        const holidays = [];
        const fn1 = jest.spyOn(holidayService, 'getHolidays').mockImplementation(() => of(holidays));

        const o = new Order();
        o.productStockData = [];
        let d = new StockData();
        d.ship_date = moment('2018-09-01').toDate();
        o.productStockData.push(d);
        d = new StockData();
        d.ship_date = moment('2018-09-15').toDate();
        o.productStockData.push(d);
        component.order = o;

        component.getExpectedDelivery();
        expect(component.expectedDeliveryDate).toBeTruthy();
        // 15/09 is saturday, 17/09 monday
        expect(component.expectedDeliveryDate.format('YYYY-MM-DD')).toBe('2018-09-17');

        holidays.push({date: moment('2018-09-17')});
        component.getExpectedDelivery();
        expect(component.expectedDeliveryDate.format('YYYY-MM-DD')).toBe('2018-09-18');

        // check when no product stock data
        component.order.productStockData = [];
        const cust = new Customer();
        cust.code = 'xxx';
        component.customer = cust;
        component.order.audits = [
            {audit_date : moment('2018-09-01').toDate(), audit_key : '', character_value : '6'},
            {audit_date : moment('2018-09-15').toDate(), audit_key : '', character_value : '5'},
            {audit_date : moment('2018-09-19').toDate(), audit_key : '', character_value : '6'}
        ];
        component.getExpectedDelivery();
        expect(component.expectedDeliveryDate.format('YYYY-MM-DD')).toBe('2018-09-20');
        cust.analysis_codes1 = 'GMCCROSSAN';
        component.getExpectedDelivery();
        expect(component.expectedDeliveryDate.format('YYYY-MM-DD')).toBe('2018-09-21');
        cust.analysis_codes1 = 'CWALSH';
        component.getExpectedDelivery();
        expect(component.expectedDeliveryDate.format('YYYY-MM-DD')).toBe('2018-09-21');
        cust.analysis_codes1 = 'SANDERSON';
        component.getExpectedDelivery();
        expect(component.expectedDeliveryDate.format('YYYY-MM-DD')).toBe('2018-09-21');

        component.order.audits = [];
        component.getExpectedDelivery();
        expect(component.expectedDeliveryDate).toBeNull();

        fn1.mockReset();
    });

    test('getStatus', () => {

        component.order = new Order();
        component.order.status = OrderStatus.Cancelled;
        let result = component.getStatus();
        expect(result).toBe('Cancelled');
        component.order.status = OrderStatus.Delivered;
        result = component.getStatus();
        expect(result).toBe('Delivered');
        component.order.status = 6;
        result = component.getStatus();
        expect(result).toBe('Processed');
        component.order.status = 5;
        result = component.getStatus();
        expect(result).toBe('Received');

    });

    test('getAuditDate', () => {
        const o = new Order();
        o.audits = [];
        let result = component.getAuditDate(o);
        expect(result).toBeNull();

        o.audits = [];
        o.audits.push({audit_date : moment('2018-10-01').toDate(), audit_key: '', character_value: ''});
        result = component.getAuditDate(o);
        expect(moment(result).format('YYYY-MM-DD')).toBe('2018-10-01');

    });

    test('getProcessedDate', () => {
        const o = new Order();
        o.audits = [];
        let result = component.getProcessedDate(o);
        expect(result).toBeNull();

        o.audits = [];
        o.audits.push({audit_date : moment('2018-10-01').toDate(), audit_key: '', character_value: '5'});
        result = component.getProcessedDate(o);
        expect(result).toBeNull();
        o.audits[0].character_value = '6';
        result = component.getProcessedDate(o);
        expect(moment(result).format('YYYY-MM-DD')).toBe('2018-10-01');
    });

    test('navigateToCaller', () => {
        const router = TestBed.get(Router);
        const fn = jest.spyOn(router, 'navigate').mockImplementation(() => {});
        component.return_to = 'invoices';
        component.navigateToCaller();
        expect(fn.mock.calls[0][0]).toEqual(expect.arrayContaining([expect.stringContaining('invoice-history')]));
        component.return_to = 'orders';
        component.navigateToCaller();
        expect(fn.mock.calls[1][0]).toEqual(expect.arrayContaining([expect.stringContaining('order-history')]));

        fn.mockReset();
    });

    test('showForCustomer', () => {
        const c = new Customer();
        c.code = 'cust';

        const fn1 = jest.spyOn(component, 'checkParams').mockImplementationOnce(() => {});
        const fn2 = jest.spyOn(component, 'searchOrder').mockImplementationOnce(() => {});
        component.paramsChecked = false;
        component.showForCustomer(c);
        expect(component.customer_code).toBe(c.code);
        expect(fn1.mock.calls.length).toBe(1);
        expect(fn2.mock.calls.length).toBe(0);
        component.paramsChecked = true;
        component.showForCustomer(c);
        expect(fn1.mock.calls.length).toBe(1);
        expect(fn2.mock.calls.length).toBe(1);

    });

    test('impersonate box', () => {
        component.user.isTopAdmin = true;
        jest.spyOn(component, 'showForCustomer').mockImplementation(() => {});
        jest.spyOn(component, 'searchOrder').mockImplementation((code) => {});
        fixture.detectChanges();
        let impBox = fixture.debugElement.query(By.directive(ImpersonateboxComponent));
        expect(impBox).toBeTruthy();

        component.user.isTopAdmin = false;
        component.user.isBranchAdmin = true;
        fixture.detectChanges();
        impBox = fixture.debugElement.query(By.directive(ImpersonateboxComponent));
        expect(impBox).toBeTruthy();

        component.user.isBranchAdmin = false;
        fixture.detectChanges();
        impBox = fixture.debugElement.query(By.directive(ImpersonateboxComponent));
        expect(impBox).toBeNull();

        jest.resetAllMocks();
    });

    test('ui - Search button', () => {
        jest.spyOn(component, 'showForCustomer').mockImplementation(() => {});
        jest.spyOn(component, 'searchOrder').mockImplementation((code) => {});
        jest.spyOn(component, 'checkParams').mockImplementationOnce(() => {});
        fixture.detectChanges();
        expect(component.order_no.length).toBe(0);
        expect($('button:contains("Search")').prop('disabled')).toBeTruthy();
        component.order_no = 'ord';
        fixture.detectChanges();
        expect($('button:contains("Search")').prop('disabled')).toBeFalsy();

    });

    test('ui - expected delivery date', () => {
        jest.spyOn(component, 'showForCustomer').mockImplementation(() => {});
        jest.spyOn(component, 'searchOrder').mockImplementation((code) => {});
        jest.spyOn(component, 'checkParams').mockImplementationOnce(() => {});
        fixture.detectChanges();
        expect($('div.hidden-xs:contains(Expected Delivery)').length).toBe(0);

        component.expectedDeliveryDate = moment('2018-10-01');
        component.order = new Order();
        component.order.status = OrderStatus.InProgress;
        fixture.detectChanges();
        expect($('div.hidden-xs:contains(Expected Delivery)').length).toBe(1);

        expect($('span:contains(Expected Delivery)').length).toBe(2);
        let formattedData: string = $('span:contains(Expected Delivery)').first().text();

        formattedData = formattedData.replace('Expected Delivery:', '').trim();
        expect(formattedData).toBe(component.expectedDeliveryDate.format(Settings.datePattern.toUpperCase()));

    });

    test('ui - test circles', () => {

        jest.spyOn(component, 'showForCustomer').mockImplementation(() => {});
        jest.spyOn(component, 'searchOrder').mockImplementation((code) => {});
        jest.spyOn(component, 'checkParams').mockImplementationOnce(() => {});
        component.expectedDeliveryDate = moment('2018-10-01');
        component.order = new Order();
        component.order.status = OrderStatus.InProgress;
        fixture.detectChanges();

        // Test circles
        // visible-xs first
        // 1st circle Received
        expect($('div.visible-xs .circle:eq(0)').hasClass('alert-success')).toBeTruthy();
        expect($('div.visible-xs .fa-xs-holder:eq(0)').hasClass('text-success')).toBeTruthy();

        // 2nd circle Processed
        expect($('div.visible-xs .circle:eq(1)').hasClass('alert-success')).toBeTruthy();
        expect($('div.visible-xs .fa-xs-holder:eq(1)').hasClass('text-warning')).toBeTruthy();

        // 3rd circle Delivered
        expect($('div.visible-xs .circle:eq(2)').hasClass('alert-danger')).toBeTruthy();

        // hidden-xs
        // 1st circle Received
        expect($('div.hidden-xs .circle:eq(0)').hasClass('alert-success')).toBeTruthy();
        expect($('div.hidden-xs .arrow-holder:eq(0)').hasClass('text-success')).toBeTruthy();

        // 2nd circle Processed
        expect($('div.hidden-xs .circle:eq(1)').hasClass('alert-success')).toBeTruthy();
        expect($('div.hidden-xs .arrow-holder:eq(1)').hasClass('text-warning')).toBeTruthy();

        // 3rd circle Delivered
        expect($('div.hidden-xs .circle:eq(2)').hasClass('alert-danger')).toBeTruthy();

        // try with delivered
        component.order.status = OrderStatus.Delivered;
        fixture.detectChanges();

        // visible-xs first
        // 1st circle Received
        expect($('div.visible-xs .circle:eq(0)').hasClass('alert-success')).toBeTruthy();
        expect($('div.visible-xs .fa-xs-holder:eq(0)').hasClass('text-success')).toBeTruthy();

        // 2nd circle Processed
        expect($('div.visible-xs .circle:eq(1)').hasClass('alert-success')).toBeTruthy();
        expect($('div.visible-xs .fa-xs-holder:eq(1)').hasClass('text-success')).toBeTruthy();

        // 3rd circle Delivered
        expect($('div.visible-xs .circle:eq(2)').hasClass('alert-success')).toBeTruthy();

        // hidden-xs
        // 1st circle Received
        expect($('div.hidden-xs .circle:eq(0)').hasClass('alert-success')).toBeTruthy();
        expect($('div.hidden-xs .arrow-holder:eq(0)').hasClass('text-success')).toBeTruthy();

        // 2nd circle Processed
        expect($('div.hidden-xs .circle:eq(1)').hasClass('alert-success')).toBeTruthy();
        expect($('div.hidden-xs .arrow-holder:eq(1)').hasClass('text-success')).toBeTruthy();

        // 3rd circle Delivered
        expect($('div.hidden-xs .circle:eq(2)').hasClass('alert-success')).toBeTruthy();


    });

    test('ui - check order history panel', () => {
        jest.spyOn(component, 'showForCustomer').mockImplementation(() => {});
        jest.spyOn(component, 'searchOrder').mockImplementation((code) => {});
        jest.spyOn(component, 'checkParams').mockImplementationOnce(() => {});
        component.expectedDeliveryDate = moment('2018-10-01');
        const o = new Order();
        o.status = OrderStatus.Delivered;
        o.date_entered = moment('2018-08-01').toDate();
        o.audits = [];
        o.audits.push({audit_date : moment('2018-09-01').toDate(), audit_key: '', character_value: '6'});
        o.invoice_date = moment('2018-09-05').toDate();
        component.order = o;

        fixture.detectChanges();

        let orderDelivered = $('div.list-group-item:contains(Order Delivered)').text();
        expect(orderDelivered).toBeTruthy();
        orderDelivered = orderDelivered.replace('Order Delivered:', '').trim();
        expect(orderDelivered).toBe(moment(o.invoice_date).format(Settings.datePattern.toUpperCase()));

        let orderProcessed = $('div.list-group-item:contains(Order Processed)').text();
        expect(orderProcessed).toBeTruthy();
        orderProcessed = orderProcessed.replace('Order Processed:', '').trim();
        expect(orderProcessed).toBe(moment(o.audits[0].audit_date).format(Settings.datePattern.toUpperCase()));

        let orderReceived = $('div.list-group-item:contains(Order Received)').text();
        expect(orderReceived).toBeTruthy();
        orderReceived = orderReceived.replace('Order Received:', '').trim();
        expect(orderReceived).toBe(moment(o.date_entered).format(Settings.datePattern.toUpperCase()));
    });

});

