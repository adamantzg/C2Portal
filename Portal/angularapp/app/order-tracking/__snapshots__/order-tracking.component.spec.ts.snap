// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`order tracking should create 1`] = `
<order-tracking
  commonService={[Function CommonService]}
  customer_code=""
  datePattern={[Function String]}
  errorMessage=""
  holidayService={[Function HolidayService]}
  orderService={[Function OrderService]}
  orderStatuses={[Function Object]}
  order_no=""
  param=""
  paramsChecked="false"
  return_to={[Function String]}
  route={[Function ActivatedRoute]}
  router={[Function Router]}
  user={[Function User]}
  userService={[Function UserService]}
>
  <div
    class="container-fluid"
  >
    <br />
    <div
      class="row"
    >
      <div
        class="col-sm-4 col-md-1"
      >
        <button
          class="btn btn-default"
        >
          Back
        </button>
      </div>
      <div
        class="col-sm-8 col-md-11"
      >
        
      </div>
    </div>
    <div
      class="row"
      style="margin-top:-20px; margin-left:-26px; margin-right:-30px; text-align: left"
    >
      <div
        class="col-sm-12"
      >
        <div
          class="panel-default"
        >
          <div
            class="panel-heading"
            style="padding-left: 27px!important;"
          >
            <div
              class="form-inline"
            >
              <div
                class="form-group"
              >
                <label>
                  Crosswater order no*:
                </label>
                <input
                  class="form-control"
                  type="text"
                />
                <button
                  class="btn btn-default"
                >
                  Search
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-sm-12"
      >
        <div>
          *Order number can be obtained from your sales order acknowledgement
        </div>
      </div>
    </div>
    
    
  </div><app-block-ui>
    
  </app-block-ui>
</order-tracking>
`;
